lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -993,7 +993,23 @@ def sanitize_network_params(self):
                                           " a VLAN name under VM network list.")
 
             if 'name' in network and find_obj(self.content, [vim.Network], network['name']) is None:
-                self.module.fail_json(msg="Network '%(name)s' does not exist." % network)
+                netfound = False
+                host = None
+                vm = self.get_vm()
+                if vm:
+                    host = vm.summary.runtime.host
+                elif self.params['esxi_hostname']:
+                    host = self.select_host()
+
+                if host is not None:
+                    for hostnet in host.network:
+                        if hostnet.name == network['name']:
```

Comment:
I am not sure if this will work because - if `find_obj` is not able find this object that means it does not exists. If it finds then you will not hit this condition. Let me know if I am wrong.

Reply:
`find_obj` apparently only finds things that are cluster-wide. It does not find networks present on only a single host.